src/movies.js

```diff
@@ -1,23 +1,65 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors() {}
+function getAllDirectors(movies) {
+  const directorList = movies.map(element => {return element.director});
+  return directorList
+}
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies() {}
+
+function howManyMovies(movies) {
+  if ( movies.length === 0) {
+    return 0;
+  } 
+  const spielbergDrama = movies.filter(element => { 
+    return element.director === 'Steven Spielberg' && element.genre.includes('Drama');
+  });
+return spielbergDrama.length
+} 
+    
 
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage() {}
+function scoresAverage(movies) {
+  if ( movies.length === 0) {
+    return 0; }
+  const average = movies.reduce((acc, value) => {
+    return acc + value.score;
+    },0); 
+
+let scoreResult = average / movies.length
+return scoreResult
+}
 
 // Iteration 4: Drama movies - Get the average of Drama Movies
-function dramaMoviesScore() {}
+function dramaMoviesScore(movies) {
+  const dramaMovies = movies.filter(element => {
+    return element.genre.includes('Drama') ;
+ });
+ const dramaScore = dramaMovies.reduce((acc, value) => {
+   return acc + value.score;
+ },0);
+
+ let dramaResult = dramaScore / dramaMovies.length
+ return dramaResult
+
+}
```

Comment:
to solve this "should return the average of the array, a floating point number"
you can reuse the last function!
like this:
```
function dramaMoviesScore(movies) {
  const dramaMovies = movies.filter(element => {
    return element.genre.includes('Drama') ;
 });
 return scoresAverage(dramaMovies)
}
```

Reply:
and that's it! Now all of the mandatory tasks (iteration 1-4) are correctly. So all in all a good job! Keep up the good work!